index.test.ts

```diff
@@ -23,13 +23,20 @@ describe('Test no cache', () => {
         const key = `${ref.id}`
         return [`item-data:${key}:id:${ref.slug}`]
       },
-      lookupFn: (items: MyValue[], ref: MyKey) =>
-        items.find((p) => p.slug === ref.slug),
+      lookupFn: (items: ArrayLike<MyValue | Error>, ref: MyKey) => {
+        for (let i = 0; i < items.length; i++) {
+          const myValue = isMyValue(items[i])
+          if (myValue && myValue.slug === ref.slug) {
+            return myValue
+          }
+        }
+        return undefined
+      },
```

Comment:
Can't we keep using items.find() or is this more efficient?

Reply:
No, unfortunately not since the return type of the batchLoadFn is `ArrayLike` but not an actual array.

Reply:
isn't this something: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find#calling_find_on_non-array_objects ?

Reply:
Nice find! I've added it.